satpy/writers/__init__.py

```diff
@@ -257,16 +255,44 @@ def add_overlay(orig, area, coast_dir, color=(0, 0, 0), width=0.5, resolution=No
                        resolution=resolution, width=width, level=level_coast)
     cw_.add_borders(img, area, outline=color,
                     resolution=resolution, width=width, level=level_borders)
-    # Only add grid if Dlonlat is given.
-    if Dlonlat:
-        # If dlonlat (minor_latlon) is not given set it equal to the Dlonlat (major_lonlat)
-        if not dlonlat:
-            dlonlat = Dlonlat
-        cw_.add_grid(img, area, Dlonlat, dlonlat, font, write_text, fill, fill_opacity,
-                     major_outline, major_width, major_outline_opacity,
-                     minor_outline, minor_width,
-                     minor_outline_opacity, minor_is_tick,
-                     lon_placement, lat_placement)
+    # Only add grid if major_lonlat is given.
+    if grid and 'major_lonlat' in grid and grid['major_lonlat']:
+        # If minor_latlon is not given set it equal to the major_lonlat
+        if 'minor_lonlat' not in grid or not grid['minor_lonlat']:
+            grid['minor_lonlat'] = grid['major_lonlat']
+        if 'font' not in grid:
+            grid['font'] = None
+        if 'write_text' not in grid:
```

Comment:
`grid.setdefault('write_text', True)` would be cleaner. Also maybe it would be better to not redefine the defaults unless they differ from pycoast? Maybe `grid_kwargs`? Not sure.

Reply:
But how can you avoid redefining?

Somehow you can go with the default, but you need a way to set your own. Right?

So at least `setdefault` gives a hint in the code

Reply:
A user shouldn't have to look at the code to determine what the defaults are. It should be documented and if the documentation in satpy says "See pycoast's documentation for :func:`pycoast:pycoast.ContourWriter.add_grid` for more information" then it isn't too hard to find (it will be linked to, I'm not sure my syntax is correct, but close).

Reply:
I agree.

But since add_grid in pycoast does not have a kwargs argument, and I pass in the information for the grid to overlay as a dictionary, how else can this be done?

Sorry for being so slow...

Reply:
I'm not sure I understand. Can't you do `cw.add_grid(img, area, **grid)`?

Besides the change to the keyword arguments that @mraspaud proposed (and handling the conversion for that) everything else is a keyword argument: https://pycoast.readthedocs.io/en/latest/api/pycoast.html#pycoast.cw_agg.ContourWriterAGG.add_grid